src/ray/gcs/gcs_server/gcs_resource_manager.h

```diff
@@ -133,6 +133,9 @@ class GcsResourceManager : public rpc::NodeResourceInfoHandler,
   /// \param data The resource loads reported by raylet.
   void UpdateResourceLoads(const rpc::ResourcesData &data);
 
+  virtual const absl::flat_hash_map<NodeID, rpc::ResourcesData> &NodeResourceReportView()
```

Comment:
Please don't make it virtual. This is really weird. Consider using gtest friend test to update the member variable for testing.